File: leetcode/js/minimum-cost-to-reach-every-position.js
/**
 * @param {number[]} cost
 * @return {number[]}
 */
var minCosts = function (cost) {
  for (let i = 1; i < cost.length; i++) {
    if (cost[i - 1] < cost[i]) {
      cost[i] = cost[i - 1];
    }
  }

  return cost;
};

File: leetcode/rust/minimum-cost-to-reach-every-position.rs
impl Solution {
    pub fn min_costs(mut cost: Vec<i32>) -> Vec<i32> {
        for i in 1..cost.len() {
            if cost[i - 1] < cost[i] {
                cost[i] = cost[i - 1];
            }
        }

        cost
    }
}
